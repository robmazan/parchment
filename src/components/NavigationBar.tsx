import React from "react";
import { Link } from "react-router-dom";
import LoginStatus from "./LoginStatus";

const NavigationBar: React.FC = props => {
  return (
    <nav>
      <ul className="nav-menu">
        <li className="nav-menu__item">
          <Link to="/" className="nav-menu__link">
            Home
          </Link>
        </li>
        <li className="nav-menu__item">
          <LoginStatus loginURI="/login" logoutURI="/logout" />
        </li>
      </ul>
    </nav>
  );
};

export default NavigationBar;
