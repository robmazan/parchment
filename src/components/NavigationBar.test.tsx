import React from "react";
import { shallow } from "enzyme";
import NavigationBar from "./NavigationBar";

describe("NavigationBar", () => {
  it("renders a navigation menu", () => {
    const component = shallow(<NavigationBar />);
    expect(component).toMatchInlineSnapshot(`
      <nav>
        <Styled(ul)>
          <Styled(li)>
            <Styled(Link)
              to="/"
            >
              Home
            </Styled(Link)>
          </Styled(li)>
          <Styled(li)>
            <Connect(LoginStatus)
              loginURI="/login"
              logoutURI="/logout"
            />
          </Styled(li)>
        </Styled(ul)>
      </nav>
    `);
  });
});
