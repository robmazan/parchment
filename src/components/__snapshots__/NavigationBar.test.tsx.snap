// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar renders a navigation menu 1`] = `
<nav>
  <ul
    className="nav-menu"
  >
    <li
      className="nav-menu__item"
    >
      <Link
        className="nav-menu__link"
        to="/"
      >
        Home
      </Link>
    </li>
    <li
      className="nav-menu__item"
    >
      <a
        className="nav-menu__link"
        href="/login"
      >
        Login
      </a>
    </li>
    <li
      className="nav-menu__item"
    >
      <a
        className="nav-menu__link"
        href="/logout"
      >
        Logout
      </a>
    </li>
  </ul>
</nav>
`;
